core/bin/external_node/src/main.rs

```diff
@@ -128,22 +125,27 @@ async fn init_tasks(
         config.optional.miniblock_seal_queue_capacity,
     );
     task_handles.push(tokio::spawn(miniblock_sealer.run()));
+
     let pool = connection_pool.clone();
+    let mut stop_receiver_for_task = stop_receiver.clone();
     task_handles.push(tokio::spawn(async move {
-        loop {
+        while !*stop_receiver_for_task.borrow_and_update() {
             let protocol_version = pool
                 .connection()
-                .await
-                .unwrap()
+                .await?
                 .protocol_versions_dal()
                 .last_used_version_id()
                 .await
                 .map(|version| version as u16);
 
-            EN_METRICS.version[&(format!("{}", version), protocol_version)].set(1);
+            EN_METRICS.version[&(version.to_string(), protocol_version)].set(1);
 
-            tokio::time::sleep(Duration::from_secs(10)).await;
+            // Error here corresponds to a timeout w/o `stop_receiver` changed; we're OK with this.
+            tokio::time::timeout(Duration::from_secs(10), stop_receiver_for_task.changed())
+                .await
+                .ok();
```

Comment:
This and similar changes are sort of off-topic, but I got annoyed during local testing that the EN doesn't stop quickly after receiving a stop signal.